cmd/k-license/k-license.go

```
@@ -0,0 +1,259 @@
+			return opts.Run()
+		},
+	}
+
+	addCmd.Flags().StringVar(&opts.templatesDir, "templates", "../../hack/boilerplate", "directory containing license templates")
+	addCmd.Flags().StringSliceVarP(&opts.excludeDirs, "exclude", "e", excludeDirsLocations, "comma-separated list of directories to exclude")
+	addCmd.Flags().StringVar(&opts.path, "path", ".", "Defaults to Current directory")
+	addCmd.Flags().BoolVar(&opts.confirm, "confirm", false, "")
+	rootCmd.AddCommand(addCmd)
+	if err := rootCmd.Execute(); err != nil {
+		os.Exit(1)
+	}
+
+}
+func (opts *Options) Run() error {
+	//excludeList := strings.Split(opts.excludeDirs, ",")
+	files := 0
+	fileList := make([]string, 0)
+	err := filepath.WalkDir(opts.path, func(path string, info fs.DirEntry, err error) error {
+		if info.IsDir() && containsExcluded(opts.excludeDirs, info.Name()) {
+			return filepath.SkipDir
+		}
+		if !info.IsDir() && (isCodeFile(path) || isBuildFile(path)) && !isGenerateFile(path) {
+			hasLic, err := hasLicense(path)
+			if !hasLic {
+				currentYear := strconv.Itoa(time.Now().Year())
+				if opts.confirm {
+					err := addLicense(path, opts.templatesDir, currentYear)
+					if err != nil {
+						return err
+					}
+					fmt.Printf("Modified %s file\n", path)
+				}
+				fileList = append(fileList, path)
+				files++
+			}
+			return err
+
+		}
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if opts.confirm {
+		fmt.Printf("Modified %v files\n", files)
+	} else {
+		fmt.Printf("DRY RUN: No file changes will be made! To make file modifications, rerun the command with  \"--confirm\" flag\n")
+		if files == 0 {
+			fmt.Printf("All files have appropriate License Headers. No changes required.\n")
+		}
+		if files > 0 {
+			fmt.Printf("%v files will be modified to add License Headers\n", files)
+			fmt.Printf("Listing files to be modified:\n")
+			for _, file := range fileList {
+				fmt.Printf("%s\n", file)
+			}
+		}
+
+	}
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
```

Let's return the error here, `cobra` will exit with a non 0 exit code if an error is returned to `RunE`.